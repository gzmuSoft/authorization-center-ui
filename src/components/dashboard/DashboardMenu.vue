<template lang="pug">
  #dashboard-menu
    div(v-for="(menu,index) in routes" :key="index")
      v-list-item-group.py-1(v-if="!menu.children", color="primary")
        v-list-item(:to="menu.to", :key="index", link)
          v-list-item-action
            v-icon {{ menu.icon }}
          v-list-item-content
            v-list-item-title {{menu.text}}
      v-subheader.mt-4.grey--text.text--darken-1(v-else) {{menu.name}}
      dashboard-menu(:routes="menu.children")

</template>

<script lang="ts">
import { Component, Vue, Prop } from 'vue-property-decorator'

@Component
export default class DashboardMenu extends Vue {
  @Prop({ type: Array }) routes!: Array<Object>
  created () {
    console.log(this.routes)
  }
}
</script>

<style lang="stylus" scoped>

</style>
