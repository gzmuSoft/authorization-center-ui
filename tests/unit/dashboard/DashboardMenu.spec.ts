import { shallowMount } from '@vue/test-utils'
import DashboardMenu from '@/views/dashboard/DashboardMenu.vue'

describe('DashboardMenu', () => {
  it('exist routes props when passed', () => {
    const routesEmpty: Array<Object> = []
    const wrapper = shallowMount(DashboardMenu, {
      propsData: { routes: routesEmpty }
    })
    expect(wrapper.vm.$props.routes).toBe(routesEmpty)
  })
  // ---------------------------------------------------------------
  it('should render menu when routes not empty and passed', function () {
    const routes: Array<Object> = [
      { name: '系统管理', children: [ { text: '仪表盘', icon: 'mdi-view-dashboard', to: { name: 'index' } } ] }
    ]
    const wrapper = shallowMount(DashboardMenu, {
      propsData: { routes }
    })
    const menus = wrapper.find(DashboardMenu)
    expect(menus.exists()).toBeTruthy()
  })
  // ---------------------------------------------------------------
  it('should check snapshot when passed', function () {
    const wrapperEmpty = shallowMount(DashboardMenu)
    expect(wrapperEmpty).toMatchSnapshot()
    const routes: Array<Object> = [
      { name: '系统管理',
        children: [
          { text: '仪表盘', icon: 'mdi-view-dashboard', to: { name: 'index' } },
          { text: '用户管理', icon: 'mdi-view-dashboard', to: { name: 'user' } }
        ]
      }
    ]
    const wrapper = shallowMount(DashboardMenu, { propsData: { routes } })
    expect(wrapper).toMatchSnapshot()
  })
})
