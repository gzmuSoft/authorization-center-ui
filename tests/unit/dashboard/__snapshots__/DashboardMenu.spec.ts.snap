// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashboardMenu.vue should check snapshot when success 1`] = `<div id="dashboard-menu"></div>`;

exports[`DashboardMenu.vue should check snapshot when success 2`] = `
<div id="dashboard-menu">
  <div>
    <v-subheader-stub class="mt-4 grey--text text--darken-1">系统管理</v-subheader-stub>
    <dashboard-menu-stub routes="[object Object],[object Object]"></dashboard-menu-stub>
  </div>
</div>
`;
